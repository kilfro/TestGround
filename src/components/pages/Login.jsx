import React from 'react';
import {loginWithEmail, loginWithGoogle} from '../../store/actions/auth';
import {connect} from 'react-redux';
import {Button, Container, TextField} from '@material-ui/core';
import '../../styles/component/pages/login.css';
import {Redirect} from "react-router-dom";
import logo from '../../logo.png';

class Login extends React.Component {
    submitEmailLogin = (event) => {
        event.preventDefault();

        const email = document.getElementById('email').value;
        const password = document.getElementById('password').value;

        this.props.loginWithEmail(email, password);
    };

    render() {
        const {authenticated, loginWithGoogle} = this.props;
        return (
            <Container className={'login-form-container'} maxWidth={'xs'}>
                {authenticated && <Redirect to={'/'}/>}
                <img src={logo} alt={'logo'}/>
                <h3>Войдите в Ваш TestGround</h3>
                <form onSubmit={this.submitEmailLogin}>
                    <TextField id={'email'} label={'E-mail'} type={'email'} fullWidth/>
                    <TextField id={'password'} label={'Пароль'} type={'password'} fullWidth/>
                    <Button id={'login-btn'} type={'submit'} color={'primary'} variant={'contained'}
                            fullWidth className={'ground_btn'}>Войти</Button>
                    <Button href={'/registration'} fullWidth>Зарегистрироваться</Button>
                </form>
                <h4>или</h4>
                <Button size={'large'} variant="outlined" onClick={() => loginWithGoogle()}>
                    <img className={'google_logo'}
                         src={'https://www.gstatic.com/mobilesdk/160512_mobilesdk/auth_service_google.svg'}
                         alt={'google logo'}/>
                    Войти с Google
                </Button>
            </Container>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        ...state.auth
    };
};

const mapDispatchToProps = {
    loginWithEmail,
    loginWithGoogle
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);