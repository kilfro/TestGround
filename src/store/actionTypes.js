export const AUTH = {
    LOGIN_WITH_GOOGLE: 'LOGIN_WITH_GOOGLE',
    LOGIN_WITH_EMAIL: 'LOGIN_WITH_EMAIL',
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',
    LOGOUT_REQUEST: 'LOGOUT_REQUEST',
    LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
    REGISTER_REQUEST: 'REGISTER_REQUEST'
};

export const ERROR = {
    CREATE: 'CREATE_ERROR',
    DELETE: 'DELETE_ERROR'
};

export const NEW_TEST = {
    INSERT_TEST_TO_DB: 'NEW_TEST_INSERT_TEST',
    CHANGE_DESCRIPTION: 'NEW_TEST_CHANGE_DESCRIPTION',
    CREATE_UID: 'NEW_TEST_CREATE_UID',
    ADD_QUESTION: 'NEW_TEST_ADD_QUESTION',
    CHANGE_QUESTION: 'NEW_TEST_CHANGE_QUESTION',
    REMOVE_QUESTION: 'NEW_TEST_REMOVE_QUESTION',
    CLEAN_STATE: 'NEW_TEST_CLEAN_STATE',
    ADD_RESULT: 'NEW_TEST_ADD_RESULT',
    CHANGE_RESULT: 'NEW_TEST_CHANGE_RESULT',
    REMOVE_RESULT: 'NEW_TEST_REMOVE_RESULT'
};