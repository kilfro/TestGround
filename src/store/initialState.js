export default {
    error: null,
    user: {
        authenticated: false,
        photoURL: null,
        uid: null,
        displayName: null,
        token: null
    },
    newTest: {
        uid: '',
        description: {
            name: '',
            additional: '',
            password: '',
            anonymous: false,
            onlyRegistered: false,
            needPassword: false
        },
        questions: [
            {
                id: 1,
                multiple: false,
                cost: 1,
                question: '',
                options: [
                    {
                        id: 1,
                        text: '',
                        isRight: false
                    }
                ]
            }
        ],
        resultDescriptions: [
            {
                id: 1,
                min: 0,
                max: 0,
                text: ''
            }
        ]
    }
};