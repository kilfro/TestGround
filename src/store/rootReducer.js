import {combineReducers} from "redux";
import {authReducer} from "./reducers/auth";
import {errorReducer} from "./reducers/error";

export default combineReducers({
    auth: authReducer,
    error: errorReducer
});