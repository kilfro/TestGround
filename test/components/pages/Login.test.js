import React from 'react';
import Login from "../../../src/components/pages/Login";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {AUTH} from "../../../src/store/actionTypes";

describe('Login page enzyme tests', () => {
    let wrapper, store;
    const loginFunc = jest.fn();
    const loginGoogle = jest.fn();
    const mockStore = configureStore([]);

    beforeEach(() => {
        store = mockStore({
            auth: {
                authenticated: false
            }
        });

        store.dispatch = jest.fn();

        wrapper = mount(
            <Provider store={store}>
                <Login loginWithEmail={loginFunc} loginWithGoogle={loginGoogle}/>
            </Provider>
        );
    });

    afterEach(() => {
        wrapper.unmount();
    });

    it('should be correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should login with email', () => {
        wrapper.find('input#email').simulate('change', {target: {id: 'email', value: 'test@email.com'}});
        wrapper.find('input#password').simulate('change', {target: {id: 'password', value: 'password'}});

        wrapper.find('form').simulate('submit');

        expect(store.dispatch).toHaveBeenCalled();
        expect(store.dispatch).toHaveBeenCalledWith({
            type: AUTH.LOGIN_WITH_EMAIL,
            payload: {email: 'test@email.com', password: 'password'}
        });
    });

    it('should login with google', () => {
        wrapper.find('button#google-btn').simulate('click');

        expect(store.dispatch).toHaveBeenCalled();
    });
});