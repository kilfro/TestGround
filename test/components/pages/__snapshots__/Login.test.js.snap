// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should be correctly 1`] = `
<div
  className="MuiContainer-root login-form-container MuiContainer-maxWidthXs"
>
  <img
    alt="logo"
    src="test-file-stub"
  />
  <h3>
    Войдите в Ваш TestGround
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="email"
        id="email-label"
      >
        E-mail
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="email"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="password"
        id="password-label"
      >
        Пароль
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="password"
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained ground_btn MuiButton-containedPrimary MuiButton-fullWidth"
      disabled={false}
      id="login-btn"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Войти
      </span>
    </button>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-fullWidth"
      href="/registration"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Зарегистрироваться
      </span>
    </a>
  </form>
  <h4>
    или
  </h4>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <img
        alt="google logo"
        className="google_logo"
        src="https://www.gstatic.com/mobilesdk/160512_mobilesdk/auth_service_google.svg"
      />
      Войти с Google
    </span>
  </button>
</div>
`;
